components/script/fetch.rs

```diff
@@ -294,39 +314,37 @@ fn queue_deferred_fetch(
     // Step 3. Set request’s keepalive to true.
     request.keep_alive = true;
     // Step 4. Let deferredRecord be a new deferred fetch record whose request is request, and whose notify invoked is onActivatedWithoutTermination.
-    let deferred_record = Arc::new(Mutex::new(DeferredFetchRecord {
+    let deferred_record = Rc::new(DeferredFetchRecord {
```

Comment:
The record seems to be moved into the fetch group, so I don't see the need fro the rc. 

I'm guessing it's needed because of the various cloning of the vec containing records, but as per my other comments I think those can be changed into mutable operations on the record in an method of fetch group.

Reply:
The `Rc` is indeed required to clone across task boundaries. See https://github.com/servo/servo/pull/41423#discussion_r2649059959 why I don't think we can use mutable borrows atm. I can check this in a follow-up PR to see if I can remove it, but I don't think we can.